packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -0,0 +1,336 @@
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Listen,
+  Prop,
+  Watch,
+} from '@stencil/core';
+import { createFocusTrap, FocusTrap } from 'focus-trap';
+import { clearAllBodyScrollLocks, disableBodyScroll } from 'body-scroll-lock';
+import { hideOthers, Undo } from 'aria-hidden';
+
+/**
+ * @click Used to detect clicks outside of modal contents to close modal
+ * @keydown Used to detect Escape key to close modal
+ */
+@Component({
+  tag: 'va-modal',
+  styleUrl: 'va-modal.css',
+  shadow: true,
+})
+export class VaModal {
+  alertActions: HTMLDivElement;
+  closeButton: HTMLButtonElement;
+  closeButtonContainer: HTMLDivElement;
+  isVisibleDirty: boolean; // used to determine if visible has changed (workaround)
+  focusTrap: FocusTrap;
+  undoAriaHidden: Undo; // used to undo aria hidden on close
+
+  @Element() el: HTMLElement;
+
+  /**
+   * Fires when modal is closed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  closeEvent: EventEmitter;
+
+  /**
+   * Fires when primary button is clicked
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  primaryButtonClick: EventEmitter;
+
+  /**
+   * Fires when secondary button is clicked
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  secondaryButtonClick: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. Fires when a
+   * a page is selected if enable-analytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * Click outside modal will trigger closeEvent
+   */
+  @Prop() clickToClose?: boolean = false;
+
+  /**
+   * If true, analytics event won't be fired
+   */
+  @Prop() disableAnalytics?: boolean = false;
+
+  /**
+   * Title/header text for the modal
+   */
+  @Prop() modalTitle?: string;
+
+  /**
+   * Hide the close button that's normally in the top right
+   */
+  @Prop() hideCloseButton?: boolean = false;
+
+  /**
+   * Selector to explicitly specify which element should receive
+   * focus when the modal is open, if the initially focused element
+   * is not the first focusable element in the document
+   */
+  @Prop() initialFocusSelector?: string;
+
+  /**
+   * Primary button text
+   */
+  @Prop() primaryButtonText?: any;
+
+  /**
+   * Secondary button text
+   */
+  @Prop() secondaryButtonText?: any;
+
+  /*
+   * Style of modal alert - info, error, success, warning
+   */
+  @Prop() status?: 'continue' | 'error' | 'info' | 'success' | 'warning';
+
+  /**
+   * If the modal is visible or not
+   */
+  @Prop() visible: boolean = false;
+
+  @Listen('click')
+  handleClick(e: MouseEvent) {
+    if (!this.clickToClose) return;
+
+    // event.target is always the shadow host
+    // event.composedPath()[0] returns the node clicked when shadow root is open
+    if (this.visible && e.composedPath()[0] === this.el) {
+      this.handleClose(e);
+    }
+  }
```

Comment:
Can you add a comment explaining that this will close the modal if the gray overlay is clicked?